Do not create symlink

milljm opened the pull request:
No longer create symlink for

```pre
CONDA_PREFIX/bin/application-opt --> CONDA_PREFIX/application/bin/application-opt
```

This was causing an issue when MOOSE tried to discover where its shared data directory was:

```bash
$(which application-opt)/../share/data/moose
CONDA_PREFIX/share/data/moose
            ^ whoops
```

Instead, modify PATH variable as we should have been doing in the activation script routine. While the build.sh looks correct, the 'share' directory already exists, and thus was never created (we were trying to create/hijack CONDA_PREFIX/share).

TLC/shell linting on generate_recipe.sh script.

TODO: need to test Linux arch. Not sure why this was working on that platform.

Closes #30108

Commit:
Do not create symlink

No longer create symlink for

CONDA_PREFIX/bin/application-opt --> CONDA_PREFIX/application/bin/application-opt

This was causing an issue when MOOSE tried to discover where its
shared data directory was: $(which application-opt)/../share/data/moose

Instead, modify PATH variable as we should have been doing in the
activation script routine. While the build.sh looks correct, the
'share' directory already exists, and thus was never created (we were
trying to create/hijack CONDA_PREFIX/share).

TLC/shell linting on generate_recipe.sh script.

TODO: need to test Linux arch. Not sure why this was working on
that platform.

Closes #30108

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2758775/), step Docs: sync website on 7c6c768 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/30138/site)

This comment will be updated on new commits.

Commit:
Fixes for ncrc app

Rather than mess with all the recipes, I am opting to put back
'ncrc-' as the default.

Also, fix an issue with sed arguments

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2759365/), step Generate coverage on 66ddf30 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>051868</th>
      <th colspan=4>#30138 66ddf3</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.29%</td>
      <td>85.29%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>109056</td>
      <td>109055</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18809</td>
      <td>18810</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/30138/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/30138/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30138/coverage/xfem)

</details>



This comment will be updated on new commits.
